src/macros.rs

```diff
@@ -0,0 +1,316 @@
+            };
+            const TIME: $crate::NaiveTime =
+                match $crate::NaiveTime::from_hms_nano_opt($h, $min, SECS_NANOS.0, SECS_NANOS.1) {
+                    Some(t) => t,
+                    None => panic!("invalid time"),
+                };
+            DATE.and_time(TIME)
+        }
+    }};
+    ($y:literal-$m:literal-$d:literal $h:literal:$min:literal) => {{
+        #[allow(clippy::zero_prefixed_literal)]
+        {
+            const DATE: $crate::NaiveDate = match $crate::NaiveDate::from_ymd_opt($y, $m, $d) {
+                Some(d) => d,
+                None => panic!("invalid calendar date"),
+            };
+            const TIME: $crate::NaiveTime = match $crate::NaiveTime::from_hms_opt($h, $min, 0) {
+                Some(t) => t,
+                None => panic!("invalid time"),
+            };
+            DATE.and_time(TIME)
+        }
+    }};
+}
+
+/// Create a `FixedOffset` with a statically known value.
+///
+/// Supported formats are '(+|-)hour:minute' and '(+|-)hour:minute:second'.
+///
+/// We don't allow an hour-only format such as `+01`. That would also make the macro parse `+0001`
+/// as the same value, which should mean the same as `+00:01`.
+///
+/// The input is checked at compile time.
+///
+/// # Examples
+/// ```
+/// use chrono::offset;
+///
+/// assert_eq!(offset!(+01:30), offset!(+01:30:00));
+/// assert_eq!(offset!(-5:00), offset!(-5:00:00));
+/// ```
+#[macro_export]
+macro_rules! offset {
+    (+$h:literal:$m:literal:$s:literal) => {{
+        #[allow(clippy::zero_prefixed_literal)]
+        {
+            assert!($h < 24 || $m < 60 || $s < 60, "invalid offset");
+            const OFFSET: $crate::FixedOffset =
+                match $crate::FixedOffset::east_opt($h * 3600 + $m * 60 + $s) {
+                    Some(t) => t,
+                    None => panic!("invalid offset"),
+                };
+            OFFSET
+        }
+    }};
+    (-$h:literal:$m:literal:$s:literal) => {{
+        #[allow(clippy::zero_prefixed_literal)]
+        {
+            assert!($h < 24 || $m < 60 || $s < 60, "invalid offset");
+            const OFFSET: $crate::FixedOffset =
+                match $crate::FixedOffset::west_opt($h * 3600 + $m * 60 + $s) {
+                    Some(t) => t,
+                    None => panic!("invalid offset"),
+                };
+            OFFSET
+        }
+    }};
+    (+$h:literal:$m:literal) => {{
+        #[allow(clippy::zero_prefixed_literal)]
+        {
+            assert!($h < 24 || $m < 60, "invalid offset");
+            const OFFSET: $crate::FixedOffset =
+                match $crate::FixedOffset::east_opt($h * 3600 + $m * 60) {
+                    Some(t) => t,
+                    None => panic!("invalid offset"),
+                };
+            OFFSET
+        }
+    }};
+    (-$h:literal:$m:literal) => {{
+        #[allow(clippy::zero_prefixed_literal)]
+        {
+            assert!($h < 24 || $m < 60, "invalid offset");
```

Comment:
I recommend splitting these out to be more specific what value is wrong. Also, reprint the wrong value.  This helps the user quickly and easily identify the culprit.

     assert!($h < 24, "invalid hour offset {}", $h);
     assert!($m < 60, "invalid minute offset {}", $m);

Apply this comment to rest of the PR.